tools/cabana/messageswidget.cc

```diff
@@ -47,21 +38,26 @@ MessagesWidget::MessagesWidget(QWidget *parent) : QWidget(parent) {
   suppress_clear = new QPushButton();
   suppress_layout->addWidget(suppress_add);
   suppress_layout->addWidget(suppress_clear);
+
+  suppress_layout->addWidget(multiple_lines_bytes = new QCheckBox(tr("Multiple Lines Bytes"), this));
+  multiple_lines_bytes->setToolTip(tr("Display bytes in multiple lines"));
+  multiple_lines_bytes->setChecked(settings.multiple_lines_bytes);
+
   main_layout->addLayout(suppress_layout);
 
   // signals/slots
-  QObject::connect(filter, &QLineEdit::textEdited, model, &MessageListModel::setFilterString);
+  QObject::connect(header, &MessageViewHeader::filtersUpdated, model, &MessageListModel::setFilterStrings);
   QObject::connect(multiple_lines_bytes, &QCheckBox::stateChanged, [=](int state) {
     settings.multiple_lines_bytes = (state == Qt::Checked);
     delegate->setMultipleLines(settings.multiple_lines_bytes);
     view->setUniformRowHeights(!settings.multiple_lines_bytes);
-    model->sortMessages();
+    model->fetchData(); // Why?
```

Comment:
The reason I call `sortMessages` here is to reset the model to force the view to recalculate the size of the bytes column. 
This call should be necessary for messages that have not been updated by updateState（or cabana is in pause mode)

Reply:
Shall we just call begin/endresetModel then?

Reply:
Yes, it's fine to just reset the model here . This way the view will recalculate the size and  display bytes column correctly.

Reply:
Do I need to use a CAN FD route for multiline bytes? Is there a public one you've been using for testing?

Reply:
Frankly, I just added some temporary code to `MessageListModel::data` for testing:

```
 } else if (role == BytesRole && index.column() == 5) {
    if (QSet<int>({1, 3, 6, 7}).contains(index.row())) {
      QByteArray dat(64, 0);
      memcpy(dat.data(), can_data.dat.data(), can_data.dat.size());
      return dat;
    }
    return can_data.dat;
  }
```